File: Artist/app/src/main/java/com/example/artist/ui/topartistslisting/TopArtistsView.java
package com.example.artist.ui.topartistslisting;

import com.example.artist.models.Artist;

import java.util.List;

public interface TopArtistsView {
    void showProgress();

    void hideProgress();

    void updateData(List<Artist> topArtists);

    void showError();

    void showEmpty();

    void hidEmpty();

}

File: Artist/app/src/main/java/com/example/artist/ui/topartistslisting/TopArtistsView.kt
package com.example.artist.ui.topartistslisting

import com.example.artist.models.Artist

interface TopArtistsView {
    fun showProgress()
    fun hideProgress()
    fun updateData(topArtists: MutableList<Artist>)
    fun showError()
    fun showEmpty()
    fun hidEmpty()
}